src/marin/generation/pipeline.py

```diff
@@ -58,7 +71,17 @@ def _update_batch(self, batch: dict[str, Any], generated_text: list[str], prompt
     def __call__(self, batch: dict[str, Any]) -> dict[str, Any]:
         """Generate a batch of text using an LLM where the example text is in dolma format in the "text" column."""
 
-        prompts = [self.template.format(example=example) for example in batch[self.prompt_column]]
+        if isinstance(self.template, list):
```

Comment:
can we convert the template consistently to be a list at construction time, instead of at call time?